### PR TITLE
Optimization of train_test_split indices shuffling

### PR DESCRIPTION
**Pull request by Alexsandruss**
Optimization of indices shuffling for scikit-learn train_test_split function.

### CODE REVIEW

File: src/daal4py.cpp [hunk 1]
```diff
@@ -678,3 +678,13 @@ void c_train_test_split(data_or_file & orig, data_or_file & train, data_or_file
 #else
 #endif
 }
+
+void c_generate_shuffled_indices(data_or_file & idx, data_or_file & random_state)
```
Comment (PetrovKP) [inline src/daal4py.cpp:682]: don't you want to merge into function `c_train_test_split`?

**Comment by Alexsandruss** [comment]
Tests of sklearn 0.21/0.22/0.23 didn't reveal problems of optimizations

File: src/daal4py.cpp (hunk 1 above)
Comment (Alexsandruss) [reply src/daal4py.cpp:682]: In some cases optimizations of `c_generate_shuffled_indices` or `c_train_test_split` may not be used. So, to use at least one of them, functions should be separate.
